File: avail-rust/docs/extrinsics/src/balances.rs
use avail_rust::error::ClientError;

pub async fn run() -> Result<(), ClientError> {
	println!("balances_transfer_all");
	transfer_all::run().await?;
	transfer_all::clean().await;
	println!("balances_transfer_allow_death");
	transfer_allow_death::run().await?;
	println!("balances_transfer_keep_alive");
	transfer_keep_alive::run().await?;

	Ok(())
}

mod transfer_all {
	use avail_rust::{
		error::ClientError,
		transactions::{BalancesEvents, SystemEvents},
		utils::account_id_from_str,
		Keypair, Nonce, Options, SecretUri, SDK,
	};
	use core::str::FromStr;

	pub async fn run() -> Result<(), ClientError> {
		let sdk = SDK::new(SDK::local_endpoint()).await?;

		// Input
		let secret_uri = SecretUri::from_str("//Alice").unwrap();
		let account = Keypair::from_uri(&secret_uri).unwrap();
		let dest = account_id_from_str("5HGjWAeFDfFCWPsjFQdVV2Msvz2XtMktvgocEZcCj68kUMaw")?; // Eve
		let keep_alive = false;

		let options = Some(Options::new().nonce(Nonce::BestBlockAndTxPool));
		let tx = sdk.tx.balances.transfer_all(dest, keep_alive);
		let result = tx.execute_wait_for_inclusion(&account, options).await?;

		dbg!(&result);
		if let Some(event) = result.find_first_event::<BalancesEvents::Transfer>() {
			dbg!(event);
		}
		if let Some(event) = result.find_first_event::<SystemEvents::KilledAccount>() {
			dbg!(event);
		}

		Ok(())
	}

	pub async fn clean() {
		let sdk = SDK::new(SDK::local_endpoint()).await.unwrap();

		// Input
		let secret_uri = SecretUri::from_str("//Eve").unwrap();
		let account = Keypair::from_uri(&secret_uri).unwrap();
		let dest = account_id_from_str("5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY").unwrap(); // Alice
		let value = SDK::one_avail() * 900_000;
		let options = Some(Options::new().nonce(Nonce::BestBlockAndTxPool));

		let tx = sdk.tx.balances.transfer_keep_alive(dest, value);
		tx.execute_wait_for_inclusion(&account, options)
			.await
			.unwrap();
	}
}

mod transfer_allow_death {
	use avail_rust::{
		error::ClientError,
		transactions::{BalancesEvents, SystemEvents},
		utils::account_id_from_str,
		Keypair, Nonce, Options, SecretUri, SDK,
	};
	use core::str::FromStr;

	pub async fn run() -> Result<(), ClientError> {
		let sdk = SDK::new(SDK::local_endpoint()).await?;

		// Input
		let secret_uri = SecretUri::from_str("//Alice").unwrap();
		let account = Keypair::from_uri(&secret_uri).unwrap();
		let dest = account_id_from_str("5HGjWAeFDfFCWPsjFQdVV2Msvz2XtMktvgocEZcCj68kUMaw")?; // Eve
		let amount = SDK::one_avail();

		let options = Some(Options::new().nonce(Nonce::BestBlockAndTxPool));
		let tx = sdk.tx.balances.transfer_allow_death(dest, amount);
		let result = tx.execute_wait_for_inclusion(&account, options).await?;

		dbg!(&result);
		if let Some(event) = result.find_first_event::<BalancesEvents::Transfer>() {
			dbg!(event);
		}
		if let Some(event) = result.find_first_event::<SystemEvents::KilledAccount>() {
			dbg!(event);
		}

		Ok(())
	}
}

mod transfer_keep_alive {
	use avail_rust::{
		error::ClientError, transactions::BalancesEvents, utils::account_id_from_str, Keypair,
		Nonce, Options, SecretUri, SDK,
	};
	use core::str::FromStr;

	pub async fn run() -> Result<(), ClientError> {
		let sdk = SDK::new(SDK::local_endpoint()).await?;

		// Input
		let secret_uri = SecretUri::from_str("//Alice").unwrap();
		let account = Keypair::from_uri(&secret_uri).unwrap();
		let dest = account_id_from_str("5HGjWAeFDfFCWPsjFQdVV2Msvz2XtMktvgocEZcCj68kUMaw")?; // Eve
		let amount = SDK::one_avail();

		let options = Some(Options::new().nonce(Nonce::BestBlockAndTxPool));
		let tx = sdk.tx.balances.transfer_keep_alive(dest, amount);
		let result = tx.execute_wait_for_inclusion(&account, options).await?;

		dbg!(&result);
		if let Some(event) = result.find_first_event::<BalancesEvents::Transfer>() {
			dbg!(event);
		}

		Ok(())
	}
}

File: avail-js/docs/extrinsics/balances.ts
import { SDK, Keyring, Events, BN } from "../../src/sdk"

export async function run() {
  console.log("Balances_TransferAll")
  await TransferAll.run()
  await TransferAll.clean()
  console.log("Balances_TransferAllowDeath")
  await TransferAllowDeath.run()
  console.log("Balances_TransferKeepAlive")
  await TransferKeepAlive.run()
}

namespace TransferAll {
  export async function run() {
    const sdk = await SDK.New(SDK.localEndpoint())

    const account = new Keyring({ type: "sr25519" }).addFromUri("//Alice")
    const dest = "5HGjWAeFDfFCWPsjFQdVV2Msvz2XtMktvgocEZcCj68kUMaw" // Eve
    const keepAlive = false

    const tx = sdk.tx.balances.transferAll(dest, keepAlive)
    const result = await tx.executeWaitForInclusion(account)
    if (result.isErr()) throw Error(result.error.reason)
    const details = result.value

    details.printDebug()
    let event1 = details.findFirstEvent(Events.Balances.Transfer)
    if (event1 != null) {
      console.log(event1)
    }

    let event2 = details.findFirstEvent(Events.System.KilledAccount)
    if (event2 != null) {
      console.log(event2)
    }
  }

  export async function clean() {
    const sdk = await SDK.New(SDK.localEndpoint())

    const account = new Keyring({ type: "sr25519" }).addFromUri("//Eve")
    const dest = "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY" // Alice
    let value = SDK.oneAvail().mul(new BN("900000"))

    const tx = sdk.tx.balances.transferKeepAlive(dest, value)
    const result = await tx.executeWaitForInclusion(account)
    result._unsafeUnwrap()
  }
}

namespace TransferAllowDeath {
  export async function run() {
    const sdk = await SDK.New(SDK.localEndpoint())

    const account = new Keyring({ type: "sr25519" }).addFromUri("//Alice")
    const dest = "5HGjWAeFDfFCWPsjFQdVV2Msvz2XtMktvgocEZcCj68kUMaw" // Eve
    const value = SDK.oneAvail()

    const tx = sdk.tx.balances.transferAllowDeath(dest, value)
    const result = await tx.executeWaitForInclusion(account, undefined)
    if (result.isErr()) throw Error(result.error.reason)
    const details = result.value

    details.printDebug()
    let event1 = details.findFirstEvent(Events.Balances.Transfer)
    if (event1 != null) {
      console.log(event1)
    }

    let event2 = details.findFirstEvent(Events.System.KilledAccount)
    if (event2 != null) {
      console.log(event2)
    }
  }
}

namespace TransferKeepAlive {
  export async function run() {
    const sdk = await SDK.New(SDK.localEndpoint())

    const account = new Keyring({ type: "sr25519" }).addFromUri("//Alice")
    const dest = "5HGjWAeFDfFCWPsjFQdVV2Msvz2XtMktvgocEZcCj68kUMaw" // Eve
    const value = SDK.oneAvail()

    const tx = sdk.tx.balances.transferKeepAlive(dest, value)
    const result = await tx.executeWaitForInclusion(account, undefined)
    if (result.isErr()) throw Error(result.error.reason)
    const details = result.value

    details.printDebug()
    let event = details.findFirstEvent(Events.Balances.Transfer)
    if (event != null) {
      console.log(event)
    }
  }
}
